core/templates/components/state-directives/hint-editor/hint-editor.component.html

```diff
@@ -37,7 +37,7 @@
     </form>
 
     <div ng-if="$ctrl.isHintLengthExceeded()" class="oppia-length-validation-error">
-      The hint length is too long. Please use at most 500 characters.
+      The hint length is too long. Please use at most 10000 characters.
```

Comment:
I think this is a bit too long as to be unrealistic -- it will confuse creators.

I suggest keeping the original limits in the UI, or "reasonable intuitive" limits, but the frontend checking can have a higher threshold. (If a creator sees an error message even in that scenario, what they have is probably already egregious.)